src/ConditionalTokensMapping.template.ts

```diff
@@ -24,12 +24,26 @@ import { partitionCheck } from './utils/conditional-utils';
 import { bigZero } from './utils/constants';
 import { getCollateralDetails } from './utils/collateralTokens';
 import { markAccountAsSeen, requireAccount } from './utils/account-utils';
+import { calculatePositionIds } from './utils/ctf-utils';
+import { getEventKey } from './utils/getEventKey';
 
 export function handlePositionSplit(event: PositionSplit): void {
+  // - don't track splits within the market makers
   if (
     FixedProductMarketMaker.load(event.params.stakeholder.toHexString()) != null
   ) {
-    // We don't track splits within the market makers
+    return;
+  }
+
+  // - don't track splits from the NegRiskAdapter
+  //   - these are handled in the NegRiskAdapterMapping
+  // - don't track splits from the CTFExchange
+  if (
+    [
+      '{{lowercase contracts.NegRiskAdapter.address}}',
+      '{{lowercase contracts.Exchange.address}}',
```

Comment:
should we have both exchange addresses in  here?

Reply:
The NegRiskExchange won't call the CTF directly; in this case, it will call the NegRiskAdapter first.  So, the stakeholder, from the perspective of the CTF, is the NegRiskAdapter.